packages/db/src/connection.ts

```diff
@@ -0,0 +1,129 @@
+import Knex from 'knex';
+import { getSecretConnectionConfig, getEnvConnectionConfig } from './config';
+
+import { envConectionConfigObject, knexSecretConnectionConfigObject } from './types';
+
+export interface knexConnectionConfigObject {
+  host: string,
+  username: string,
+  password: string,
+  database?: string,
+}
+
+interface knexConfigObject {
+  client: string,
+  connection: knexConnectionConfigObject,
+  acquireConnectionTimeout: number,
+  asyncStackTraces: boolean,
+  debug: boolean,
+  migrations?: {
+    directory: string
+  }
+}
+
+interface knexEnvironmentObject extends NodeJS.ProcessEnv {
```

Comment:
Since `NodeJS.ProcessEnv` is already defined as:

```ts
interface ProcessEnv {
  [key: string]: string | undefined;
}
```

I'm not sure this interface is actually doing anything.

Reply:
Fair - Originally it *did* as there was a required key, however the thought process to leave it there was something along the lines of documentation-in-code.   With more sleep, that's not really useful. 

b1f60493c